File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__leet_paiza.cs
// 実行時間 30ms
using static System.Console;

internal class Program
{
    static void Main()
    {
        const string p = "paiza";
        char[] s = ReadLine().ToCharArray();

        if( string.Join( "", s ).Contains( p ) == true ) {
            WriteLine( p );
            return;
        }
        foreach( int i in System.Linq.Enumerable.Range( 0, s.Length ) ) {
            if( s[i] == '4' || s[i] == '@' ) {
                s[i] = 'a';
            } else if( s[i] == '1' || s[i] == '!' ) {
                s[i] = 'i';
            } else if( s[i] == '2' ) {
                s[i] = 'z';
            }
        }
        WriteLine( string.Join( "", s ).Contains( p ) ? "leet" : "nothing" );
        return;
    }
}

File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__leet_paiza.py
# 実行時間 20ms
import sys

P = "paiza"
s = []

for c in input():
    s.append(c)
if P in "".join(s):
    print(P)
    sys.exit()
for i in range(len(s)):
    if s[i] == "4" or s[i] == "@":
        s[i] = "a"
    elif s[i] == "1" or s[i] == "!":
        s[i] = "i"
    elif s[i] == "2":
        s[i] = "z"
print("leet" if P in "".join(s) else "nothing")
